### PR TITLE
Lambda leaderboard loading

### LINKED ISSUE #6: Move leaderboard fetching to a Lambda Function
Write a lambda function that will run on a CRON schedule and write to the managed DB.

### PR DESCRIPTION
**Pull request by BoraxTheClean**
Resolves #6 

This is mostly boilerplate to move the Leaderboard fetching over to Lambda. 

The real work comes when we:
- Choose a datastore.
- Integrate the current deployment to use the new datastore.

### CODE REVIEW

File: .github/workflows/aws.yml [hunk 1]
```diff
@@ -16,10 +20,10 @@ jobs:
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_ACCESS_KEY }}
-          aws-region: us-east-1
+          aws-region: ${{ secrets.REGION }}
       - name: Package and Deploy to AWS CloudFormation
         run: |
-          aws cloudformation package --template-file lambda-loader/template.yaml \
+          aws cloudformation package --template-file ./lambda-loader/template.yaml \
```
Comment (BoraxTheClean) [inline .github/workflows/aws.yml:30]:
```suggestion
          mkdir dist
          cp -r src/ dist/
          aws cloudformation package --template-file ./lambda-loader/template.yaml \
```
@JimLiu0 If it helps

**Comment by BoraxTheClean** [comment]
@JimLiu0 After badb8310a63a746c027822d0dd6cacf936a7d853, GH Actions are failing because the stack is failing to deploy. After a quick glance at the template I can't tell why it is failing. 

Please advise as to what the error is and delete the failed stack.

File: .github/workflows/aws.yml [hunk 2]
```diff
@@ -16,26 +20,17 @@ jobs:
...
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_ACCESS_KEY }}
-          aws-region: us-east-1
+          aws-region: ${{ secrets.REGION }}
       - name: Package and Deploy to AWS CloudFormation
         run: |
-          aws cloudformation package --template-file lambda-loader/template.yaml \
+          mkdir -p lambda-loader/dist
+          cp -r lambda-loader/src/ lambda-loader/dist/
+          pip3 install -r lambda-loader/requirements.txt -t lambda-loader/dist
```
Comment (BoraxTheClean) [inline .github/workflows/aws.yml]:
```suggestion
          pip3 install -r lambda-loader/requirements.txt -t lambda-loader/dist/
```

**Comment by BoraxTheClean** [comment]
@JimLiu0 Feel free to keep running with this. I kind of left this half done but I'm gonna crash. e192abda14521ec2cc6991739ee6daeeb449db48.

File: lambda-loader/src/api.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ def getLeaderboardSnapshot():
         accounts = json.loads(r.text)['leaderboard']['rows']
 
         for account in accounts:
-            name = account['accountid'].encode('utf-8').lower()
+            name = account['accountid'].encode('utf-8').lower() # Why is this converted to bytes?
```
Comment (JimLiu0) [inline lambda-loader/src/api.py:19]: I believe I needed it to handle some of the asian language names but maybe it's not actually needed.

File: lambda-loader/src/getMidnight.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+from pytz import timezone
+
+def getMidnightTTL():
+    tz = timezone('US/Pacific')
+    currentDate = date.today()
+    midnight_without_tzinfo = datetime.combine(currentDate, time())
+    midnight_with_tzinfo = tz.localize(midnight_without_tzinfo)
+    midnight_as_epoch = int(midnight_with_tzinfo.timestamp())
+
+    return midnight_as_epoch
```
Comment (BoraxTheClean) [inline lambda-loader/src/getMidnight.py]: We could probably move this into the data class and make the method private `_getMidnightTTL`

File: lambda-loader/src/api.py (hunk 3 above)
Comment (BoraxTheClean) [reply lambda-loader/src/api.py:19]: Yeah I am curious if there are some other bytes operations down the road on request workflows.

DynamoDB can of course accept strings with any utf-8 characters, so strange characters can be part of primary keys.

If we need for it to be bytes, we need to recreate the database and change the type of the primary key from string to bytes.

**Comment by BoraxTheClean** [comment]
From discord:

- [x]  Cleaning up the code
- [ ]  making sure stuff isn’t missing from leaderboardbot.py in handler
   - I'm not sure what this means
- [ ] Fix duplicate/stale data issues.
   - This should be done in a different PR, we should just add time stamps to the data model.
- [x] Performance tuning lambda (Ideally an invocation would take under a minute)
- [ ] Validate DynamoDB is behaving as expected presently @JimLiu0 
- [ ] Make sure we are staying under our provisioned capacity for dynamo table. (My iam user can't view them)
- [ ] Use dynamo batch apis?

**Comment by BoraxTheClean** [comment]
At 384 MB a cold start takes 33 seconds.
At 512 MB a cold start takes 28 seconds.

I'll leave it at 384 right now.

**Pull request merged**